request-server.go

```diff
@@ -288,9 +288,9 @@ func cleanPath(p string) string {
 }
 
 func cleanPathWithBase(base, p string) string {
-	p = filepath.ToSlash(p)
+	p = filepath.ToSlash(filepath.Clean(p))
 	if !path.IsAbs(p) {
 		return path.Join(base, p)
 	}
-	return path.Clean(p)
+	return p
```

Comment:
I did some basic tests and all worked, anyway since `path.Join` calls Clean internally, I'll leave path.Clean(p) here too

Reply:
Hm… the point here in removing this `path.Clean()` is that PuTTY sftp sanitizes names with `FXP_REALPATH` and this removes the double slash at the start of a `//UNC/Path` this means that while on my Linux machine, I could do `ls //wsl$/openSUSE-Leap-15.3` attempting to do the same on my PuTTY sftp client, resulted in the path being cleaned in the `FXP_REALPATH`, and then failing with `cannot open: /wsl$/openSUSE-Leap-15.3`

Reply:
So we could have the same issue if the path is not absolute and we hit `return path.Join(base, p)`

Reply:
True, but this should only apply in the case where the base being added is a UNC address. So, for example `cleanPathWithBase("//wsl$/openSUSE-Leap-15.3", aPath)`

I think in this case, we’re kind of a bit stuck on “having a non-default base paths is not particularly robust and a little bit of a hack”. 🤔 For this particular case, I’m a bit more willing to let the inconsistent behavior slide.

Reply:
Allowing non cleaned paths could break some usages, let's merge this patch and see if someone complaints. I'll try to do some tests myself using wsl (I never tested it) as soon as I have some free time, it could take a while

Reply:
Right, this is why I think keeping the `path.Join` which will auto-clean as well is the better choice for now, even though it would break UNC base paths. Getting some sort of logic to correctly handle all the nuances of properly rooting all reasonable base/path combos would be 😩 .

It’s at least pretty nice that we can do UNC paths with a “local” datastore with WSL, if I had to setup a remote SMB server to test any of this, I might very well tear my hair out. 😆